apps/frontend/src/modules/courses/slices/actions.ts

```diff
@@ -0,0 +1,21 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+import { type AsyncThunkConfig } from "~/libs/types/types.js";
+
+import {
+	type CourseSearchFilterDto,
+	type CoursesResponseDto,
+} from "../libs/types/types.js";
+import { name as sliceName } from "./courses.slice.js";
+
+const search = createAsyncThunk<
```

Comment:
```suggestion
const getAll = createAsyncThunk<
```
because `getAll` always will be a search in the future